packages/launchpad/cypress/e2e/integration/choose-a-browser.spec.ts

```diff
@@ -0,0 +1,260 @@
+      cy.get('[data-cy="alert-suffix-icon"]').click()
+      cy.get('[data-cy="alert-header"]').should('not.exist')
+    })
+
+    it('shows installed browsers with their relevant properties', () => {
+      cy.openProject('launchpad', ['--e2e'])
+
+      cy.visitLaunchpad()
+
+      stepThroughConfigPages()
+
+      cy.get('h1').should('contain', 'Choose a Browser')
+
+      const { chrome, firefox, electron, edge } = getBrowserAliases()
+
+      cy.get(chrome)
+      .should('contain', 'Chrome')
+      .and('contain', 'v1.x')
+      .find('img')
+      .should('have.attr', 'alt', 'Chrome')
+
+      cy.get(firefox)
+      .should('contain', 'Firefox')
+      .and('contain', 'v2.x')
+      .find('img')
+      .should('have.attr', 'alt', 'Firefox')
+
+      cy.get(electron)
+      .should('contain', 'Electron')
+      .and('contain', 'v3.x')
+      .find('img')
+      .should('have.attr', 'alt', 'Electron')
+
+      cy.get(edge)
+      .should('contain', 'Edge')
+      .and('contain', 'v4.x')
+      .find('img')
+      .should('have.attr', 'alt', 'Edge')
+    })
+
+    it('performs mutation to launch selected browser when launch button is pressed', () => {
+      cy.openProject('launchpad', ['--e2e'])
+
+      cy.visitLaunchpad()
+
+      stepThroughConfigPages()
+
+      cy.get('h1').should('contain', 'Choose a Browser')
+
+      cy.contains('Launch Chrome').as('launchButton')
+
+      // Stub out response to prevent browser launch but not break internals
+      cy.intercept('mutation-OpenBrowser_LaunchProject', {
+        body: {
+          data: {
+            launchOpenProject: true,
+            setProjectPreferences: {
+              currentProject: {
+                id: 'test-id',
+                title: 'launchpad',
+                __typename: 'CurrentProject',
+              },
+              __typename: 'Query',
+            },
+          },
+        },
+      }).as('launchProject')
+
+      cy.get('@launchButton').click()
+
+      cy.wait('@launchProject').then(({ request }) => {
+        expect(request?.body.variables.browserPath).to.contain('/test/chrome/path')
+        expect(request?.body.variables.testingType).to.eq('e2e')
+      })
+    })
+
+    it('performs mutation to change selected browser when browser item is clicked', () => {
+      cy.openProject('launchpad', ['--e2e'])
+
+      cy.visitLaunchpad()
+      stepThroughConfigPages()
+
+      cy.get('h1').should('contain', 'Choose a Browser')
+
+      const { chrome, firefox } = getBrowserAliases()
+
+      cy.get(chrome).should('have.attr', 'data-selected-browser', 'true')
+      cy.get(firefox).should('have.attr', 'data-selected-browser', 'false')
+
+      cy.contains('Launch Chrome').should('be.visible')
+
+      cy.intercept('mutation-OpenBrowserList_SetBrowser').as('setBrowser')
+
+      cy.get(firefox).click().then(($element) => {
+        cy.wait('@setBrowser').its('request.body.variables.id').should('eq', $element.attr('data-browser-id'))
```

Comment:
Feedback wanted: does this kind of assertion feel too granular for an e2e test?

Reply:
I feel like I go back & forth. I think here this could be covered by a component test